rock-paper-scissors/scripts/rock-paper-scissors.js

```diff
@@ -0,0 +1,158 @@
+function computerPlay() {
+  computerSelection = gameChoices[Math.floor(Math.random() * gameChoices.length)];
+  return computerSelection;
+}
+// playGame takes the player and computer input. passes computerPlay argument to the correct
+//scoring function.
+
+function playGame(playerSelection, computerSelection) {
+  if (playerSelection === "rock") {
+    rock(computerSelection);
+  } else if (playerSelection === "paper") {
+    paper(computerSelection);
+  } else if (playerSelection === "scissors") {
+    scissors(computerSelection);
+  }
+}
+
+// "add" functions tally the score. called by functions rock(), paper(), scissors().
+function addPlayerWins(playerSelection, computerSelection) {
+  console.log(`You win! ${playerSelection} beats ${computerSelection}`);
+  roundCount += 1;
+  playerWins += 1;
+}
+function addComputerWins(playerSelection, computerSelection) {
+  console.log(`You Lose! ${computerSelection} beats ${playerSelection}!`);
+  roundCount += 1;
+  computerWins += 1;
+}
+function addNobodyWins() {
+  console.log(`This round was a draw! Try again!`);
+  roundCount += 1;
+  nobodyWins += 1;
+}
+
+//rock() paper() and scissors () take computerSelection as argument.
+// these decides the outcome of the round amd call the add functions.
+function rock(computerSelection) {
+  if (computerSelection === "rock") {
+    addNobodyWins();
+  } else if (computerSelection === "paper") {
+    addPlayerWins(playerSelection, computerSelection);
+  } else if (computerSelection === "scissors") {
+    addComputerWins(playerSelection, computerSelection);
+  }
+  return roundCounter(roundCount);
+}
+
+function paper(computerSelection) {
+  if (computerSelection === "rock") {
+    addComputerWins(playerSelection, computerSelection);
+  } else if (computerSelection === "paper") {
+    addNobodyWins();
+  } else if (computerSelection === "scissors") {
+    addPlayerWins(playerSelection, computerSelection);
+  }
+  return roundCounter(roundCount);
+}
+
+function scissors(computerSelection) {
+  if (computerSelection === "rock") {
+    addComputerWins(playerSelection, computerSelection);
+  } else if (computerSelection === "paper") {
+    addPlayerWins(playerSelection, computerSelection);
+  } else if (computerSelection === "scissors") {
+    addNobodyWins();
+  }
+  return roundCounter(roundCount);
+}
+
+// roundCounter checks to see if the match score equals a win or loss.
+function newGamePrompt() {
+  let playAgain = prompt("Would you like to play again? Yes or No?").toLowerCase();
+  return playAgain;
+}
+
+function newGameNo() {
+  console.log("Thank you for playing!");
+  globalReset();
+}
+
+function newGameYes() {
+  globalReset();
+  startNewRound();
+}
+
+function roundCounter(roundCount) {
+  if (roundCount < 5) {
+    if (playerWins === 3) {
+      console.log("You are the winner!");
+      playAgain = newGamePrompt();
+      if (playAgain === "yes") {
+        newGameYes()
+      } else if (playAgain === "no") {
+        newGameNo()
+      }
```

Comment:
As with a comment above, it's useful to treat `if` statements like mathematical equations. We can see that this block of code is repeated 3 times. Already we can extract that to a function to make it easier to read e.g.
```
function playAgain() {
    const playAgain = newGamePrompt();
      if (playAgain === "yes") {
        newGameYes()
      } else if (playAgain === "no") {
        newGameNo()
      }
}
```

But then we are left with
```
 if (playerWins === 3) {
      console.log("You are the winner!");
      playAgain();
} else if  (computerWins === 3) {
       console.log("You lose!");
       playAgain();
} else if (nobodyWins === 3) {
        console.log("Nobody won!");
       playAgain();
} else {
        startNewRound();
}
```

And I think we can try to figure out a way to extract `playAgain()` from this just like a mathematical equation. I'll leave that up as an exercise for you to figure out though! Hint: You'll likely need a new wrapping `if` statement.

As an aside, I feel like this code can't work? `playAgain` isn't declared anywhere in the scope of this function?